worlds/mlss/Regions.py

```diff
@@ -0,0 +1,128 @@
+    create_region(world, "Surfable", surfable, excluded)
+    create_region(world, "Hooniversity", hooniversity, excluded)
+    create_region(world, "GwarharEntrance", gwarharEntrance, excluded)
+    create_region(world, "GwarharMain", gwarharMain, excluded)
+    create_region(world, "TeeheeValley", teeheeValley, excluded)
+    create_region(world, "Winkle", winkle, excluded)
+    create_region(world, "Sewers", sewers, excluded)
+    create_region(world, "Airport", airport, excluded)
+    create_region(world, "JokesEntrance", jokesEntrance, excluded)
+    create_region(world, "JokesMain", jokesMain, excluded)
+    create_region(world, "PostJokes", postJokes, excluded)
+    create_region(world, "Theater", theater, excluded)
+    create_region(world, "Fungitown", fungitown, excluded)
+    create_region(world, "FungitownBeanstar", fungitownBeanstar, excluded)
+    create_region(world, "FungitownBirdo", fungitownBirdo, excluded)
+    create_region(world, "BooStatue", booStatue, excluded)
+    create_region(world, "Oasis", oasis, excluded)
+    create_region(world, "BaseUltraRocks", baseUltraRocks, excluded)
+
+    if world.options.coins:
+        create_region(world, "Coins", coins, excluded)
+
+    if not world.options.castle_skip:
+        create_region(world, "Bowser's Castle", bowsers, excluded)
+        create_region(world, "Bowser's Castle Mini", bowsersMini, excluded)
+
+
+def connect_regions(world: "MLSSWorld"):
+    names: typing.Dict[str, int] = {}
+
+    connect(world, names, "Menu", "Main Area")
+    if world.options.coins:
+        connect(world, names, "Main Area", "Coins")
+    connect(world, names, "Main Area", "BaseUltraRocks", lambda state: StateLogic.ultra(state, world.player))
+    connect(world, names, "Main Area", "Chucklehuck Woods", lambda state: StateLogic.brooch(state, world.player))
+    connect(world, names, "Main Area", "BooStatue", lambda state: StateLogic.canCrash(state, world.player))
+    connect(world, names, "Main Area", "Hooniversity", lambda state: StateLogic.canDig(state, world.player) and StateLogic.canMini(state, world.player))
+    connect(world, names, "Hooniversity", "Oasis")
+    connect(world, names, "Main Area", "TeeheeValley", lambda state: StateLogic.super(state, world.player) or StateLogic.canDash(state, world.player))
+    connect(world, names, "TeeheeValley", "GwarharEntrance", lambda state: StateLogic.membership(state, world.player) and StateLogic.fire(state, world.player))
+    connect(world, names, "TeeheeValley", "Oasis", lambda state: StateLogic.membership(state, world.player) and StateLogic.fire(state, world.player))
+    connect(world, names, "TeeheeValley", "Fungitown", lambda state: StateLogic.thunder(state, world.player) and StateLogic.castleTown(state, world.player) and StateLogic.rose(state, world.player))
+    connect(world, names, "Fungitown", "FungitownBeanstar", lambda state: StateLogic.pieces(state, world.player) or state.can_reach("FungitownBirdo", "Region", world.player))
```

Comment:
Using `state.can_reach(..., "Region", ...)` from an Entrance access condition produces non-deterministic behavior and will cause random generation failures. 

You must add the following line each time you do it:
`world.multiworld.register_indirect_condition(region, entrance)`